grunt/helpers/Translate.js

```diff
@@ -0,0 +1,375 @@
+
+/**
+ * @typedef {import('./Framework')} Framework
+ */
+
+/**
+ * Pulls together schemas and data to enable both the export and import of data item attributes
+ * marked by the schemas as translatable.
+ */
+class Translate {
+
+  /**
+   * @param {Object} options
+   * @param {Framework} options.framework
+   * @param {function} options.includedFilter
+   * @param {string} options.masterLang
+   * @param {string} options.targetLang
+   * @param {string} options.format
+   * @param {string} options.csvDelimiter
+   * @param {boolean} options.shouldReplaceExisting
+   * @param {string} options.jsonext
+   * @param {string} options.sourcePath
+   * @param {string} options.languagePath
+   * @param {string} options.outputPath
+   * @param {boolean} options.isTest
+   * @param {function} options.log
+   */
+  constructor({
+    framework = null,
+    includedFilter = function() { return true; },
+    masterLang = 'en',
+    targetLang = null,
+    format = 'csv',
+    csvDelimiter = ',',
+    shouldReplaceExisting = false,
+    jsonext = 'json',
+    sourcePath = '',
+    languagePath = path.join(process.cwd(), 'languagefiles'),
+    outputPath = '',
+    useOutputData = false,
+    isTest = false,
+    log = console.log,
+    warn = console.warn
+  } = {}) {
+    /** @type {Framework} */
+    this.framework = framework;
+    /** @type {function} */
+    this.includedFilter = includedFilter;
+    /** @type {string} */
+    this.masterLang = masterLang;
+    /** @type {string} */
+    this.targetLang = targetLang;
+    // format can be raw, json or csv
+    /** @type {string} */
+    this.format = format;
+    /** @type {string} */
+    this.csvDelimiter = csvDelimiter;
+    /** @type {boolean} */
+    this.shouldReplaceExisting = shouldReplaceExisting;
+    /** @type {string} */
+    this.jsonext = jsonext;
+    /** @type {string} */
+    this.sourcePath = sourcePath.replace(/\\/g, '/');
+    /** @type {string} */
+    this.outputPath = outputPath.replace(/\\/g, '/');
+    /** @type {Framework} */
+    this.useOutputData = useOutputData;
+    /** @type {string} */
+    this.languagePath = languagePath.replace(/\\/g, '/');
+    /** @type {Data} */
+    this.data = null;
+    /** @type {boolean} */
+    this.isTest = isTest;
+    /** @type {function} */
+    this.log = log;
+    /** @type {function} */
+    this.warn = warn;
+  }
+
+  /** @returns {Translate} */
+  load() {
+    this.data = new Data({
+      framework: this.framework,
+      sourcePath: this.useOutputData ? this.outputPath : this.sourcePath,
+      jsonext: this.jsonext,
+      log: this.log
+    });
+    this.data.load();
+    return this;
+  }
+
+  /**
+   * Produces a single JSON file or a series of CSV files representing all of the
+   * files and file items in the data structure.
+   * @returns {Translate}
+   */
+  async export() {
+
+    const schemas = new Schemas({ framework: this.framework, includedFilter: this.includedFilter, sourcePath: this.sourcePath });
```

Comment:
bit more destructuring could be done here-

`new Schemas({ framework, includedFilter, sourcePath } = this)`

Reply:
I did think about this, and figured it was perhaps a little too concise and bleedy for my tastes, although perfectly valid and concise.

Reply:
Tom G has a genius solution, but it's not clear what's going on at a glance:
```js
(({ a, b, c }) => ({ a, b, c }))({ a: 1, b: 2, c: 3, d: 4 })
```
```js
functionReturnValue = ArrowFunction((argsPickDestructorVariableCreation) => return(newObjectAssignmentFromVariables))CallArrowFunction(input)
```